src/mbpfan.c

```diff
@@ -370,29 +390,30 @@ t_sensors *refresh_sensors(t_sensors *sensors)
     return sensors;
 }
 
-
-/* Controls the speed of the fan */
-void set_fan_speed(t_fans* fans, int speed)
+/* Controls the speed of a fan */
+void set_fan_speed(t_fans* fan, int speed)
 {
-    t_fans *tmp = fans;
-
-    while(tmp != NULL) {
-        if(tmp->file != NULL && tmp->old_speed != speed) {
-            char buf[16];
-            int len = snprintf(buf, sizeof(buf), "%d", speed);
-            int res = pwrite(fileno(tmp->file), buf, len, /*offset=*/ 0);
-            if (res == -1) {
-                perror("Could not set fan speed");
-            }
-            tmp->old_speed = speed;
-        }
-
-        tmp = tmp->next;
+    if(fan != NULL && fan->file != NULL && fan->old_speed != speed) {
+       char buf[16];
+       int len = snprintf(buf, sizeof(buf), "%d", speed);
+       int res = pwrite(fileno(fan->file), buf, len, /*offset=*/ 0);
+       if (res == -1) {
+          perror("Could not set fan speed");
+       }
+       fan->old_speed = speed;
     }
-
 }
 
+void set_fan_minimum_speed(t_fans* fans)
+{
+   printf("set_fan_minimum_speed called\n");
```

Comment:
Needed?

Reply:
Thanks for the feedback!, I guess the printf-statement. No.

Reply:
> This approach seems fine. However I cannot run the unit tests on a MacBook Air:
> 
> ```
> $ sudo bin/mbpfan -t
> Starting the tests..
> It is normal for them to take a bit to finish.
> Error: max_fan_speed value is not 6200 
> Tests run: 5
> ```
My skills in git/GitHub is both new and not extensive: I had missed to add the new test-configs. It should run better now.

I also assume I do not need to do any update on the pull request? I have committed the changes to the fork. That should be enough right?